src/webapi/timer_future.rs

```diff
@@ -0,0 +1,205 @@
+use std::sync::{Arc, Mutex};
+use webcore::once::Once;
+use webcore::value::Value;
+use webapi::error::Error;
+use futures_core::{Future, Poll, Async};
+use futures_core::task::{Waker, Context};
+use futures_core::stream::Stream;
+use futures_channel::oneshot;
+
+
+#[inline]
+fn convert_to_i32( ms: u32 ) -> i32 {
+    let ms: i32 = ms as i32;
+
+    assert!( ms >= 0, "ms must be less than 2147483648" );
+
+    ms
+}
+
+
+/// The [`Future`](https://docs.rs/futures/0.2.*/futures/future/trait.Future.html) which is returned by
+/// [`wait`](fn.wait.html).
+// This isn't implemented as a PromiseFuture because Promises do not support cancellation
+#[derive( Debug )]
+pub struct Wait {
+    receiver: oneshot::Receiver< () >,
+    timer: Value,
+}
+
+impl Wait {
+    fn new( ms: u32 ) -> Self {
+        // We accept a u32 because we don't want negative values, however setTimeout requires it to be i32
+        let ms = convert_to_i32( ms );
+
+        let ( sender, receiver ) = oneshot::channel();
+
+        let callback = move || {
+            // TODO is this correct ?
+            match sender.send( () ) {
+                Ok( _ ) => {},
+                Err( _ ) => {},
+            };
```

Comment:
It doesn't really matter but I guess `let _ = sender.send(())` would be prettier than a `match`.